postgen.js

```diff
@@ -198,7 +198,10 @@ export function filterBom(bomJson, options) {
       newcomponents.push(newPkgMap[aref]);
     }
     for (const adep of bomJson.dependencies) {
-      if (newPkgMap[adep.ref]) {
+      if (
+        newPkgMap[adep.ref] ||
+        adep.ref === bomJson.metadata.component["bom-ref"]
```

Comment:
This is a good fix! Since metadata.component could have child `components`, can we do this by populating the newPkgMap in advance by adding both metadata.component and metadata.component.components?

Reply:
Good point. The new version should do the trick.